lib/project_types/script/layers/infrastructure/languages/assemblyscript_task_runner.rb

```diff
@@ -5,17 +5,29 @@ module Layers
     module Infrastructure
       module Languages
         class AssemblyScriptTaskRunner < TaskRunner
+          include ToolVersionChecker
+          TOOLS = {
+            "node" => {
+              "minimum_version" => "14.15.0",
+            },
+            "npm" => {
+              "minimum_version" => "5.2.0",
+            },
+          }
+
```

Comment:
I see Stringly typing being used to communicate with `ToolVersionChecker`.

Couple Qs:

1. Any reason for making `ToolVersionChecker` a mixin? Sounds like we could call it directly instead. 
2. Instead of sending a string to verify a tool, have you considered making multiple methods on `ToolVersionChecker`? e.g. `ToolVersionChecker.npm(min_npm_version)`